### ISSUE TITLE
Child queries for @deferred components are not properly reset when component is removed

### ISSUE THREAD

**Issue by leonsenft** [dart-lang/angular#1540]
Consider the following example:

```dart
@Component(
  selector: 'example',
  template: '''
    <div *ngIf="show">
      <expensive @deferred></expensive>
    </div>
  ''',
  directives: [ExpensiveComponent, NgIf],
)
class ExampleComponent {
  @ViewChild(ExpensiveComponent)
  var deferredChild;

  bool get show => ...;
}
```

The `deferredChild` query is properly set with the deferred component when loaded; however, it's not reset to null when the enclosing structural directive removes the deferred component.

**Comment by matanlurey** [comment]
@leonsenft I am assuming this could be a known issue and fixed in 5.1/5.0.1, right?

**Comment by matanlurey** [comment]
@leonsenft My _guess_ is this is the same issue as https://github.com/dart-lang/angular/issues/1539?

**Comment by leonsenft** [comment]
Possibly, but IIRC I think we just don't generate the code to invalidate queries in this case.

**Comment by leonsenft** [comment]
Oh but this example uses the correct template, unlike the issue #1539.

**Comment by matanlurey** [comment]
Ah OK, good point. I'll take a look after #1539 in any case.

**Comment by matanlurey** [comment]
I'm hoping this is just https://github.com/dart-lang/angular/issues/1539, but I'll validate after.

**Comment by matanlurey** [comment]
This does work after #1539, and I'm adding a test in https://github.com/matanlurey/angular/commit/b328544ee188eb726bbc36e751a7e183f028b150.
